### DISCUSSION TITLE
Velocity on hematopoiesis: is it advisable at all?

### DISCUSSION THREAD

**Discussion by GabrielBaldissera** [theislab/scvelo#430] [Q&A]
Hello!

I am working in the velocity analysis and I am wondering wether I should trust my results or not. My dataset is composed of cells involved in zebrafish hematopoiesis at 26hpf. The overall directions I see are not according to what I would expect for this process.

The following is the visualization for my WT sample. 
Clusters 0, 1, 2, and 7  are all artery cells. Cluster 3 is a group of endothelial cells called HemECs (Hemogenic endothelial cells); cluster 4 and 6 are nHSPCs (Nascent Hemogenic stem and progenitor cells) and clusters 5, 8 and 9 are groups of progenitor cells (Erytroid, neutrophils and macrophages respectvely).
Therefore, as the endothelial-to-hematopoietic transition (EHT) is going on, we would expect the summarized velocities to be in the other direction (0,2->1->3->4,6->5,8,9).
I have tried running the analysis with different models and the results were very similar (Dynamic worked better to visualize phase portraits).

![image](https://user-images.githubusercontent.com/47226253/115313653-aa79fe80-a141-11eb-984a-47ac1828b3a3.png)

The directions 4,6->5,8,9 are according to what I would expect but not the rest.

I have read  #173 and therefore I did some of the check-up plots mentioned. Here is the  fragment ratio plot as suggested by @VolkerBergen:
![image](https://user-images.githubusercontent.com/47226253/115319946-e5366380-a14e-11eb-9c75-79931079b1f8.png)

Here is the confidence plot:
![image](https://user-images.githubusercontent.com/47226253/115318044-df3e8380-a14a-11eb-8a3b-c4da1783398e.png)

Here is the scvelo proportion plot:
![image](https://user-images.githubusercontent.com/47226253/115312985-74884a80-a140-11eb-8356-01da48d63ee6.png)

Based on the above plots, I could not identify any issue with the data.

Next thing I did was to look to phase portraits of individual genes. I have plotted 3 markers that we use to identify EHT cells. Myb is the marker of nHSPCs (Cluster 4 and 6), runx1 is the marker of HemECs (Cluster 3) and gata2b is the marker for artery cells that will give rise to HemECs (Cluster 1). Based on the following immage, the velocities are according to expected. That makes me wonder if there is another biological process that is "masking" EHT or if the fact that this process takes a long time is responsible for not capturing it in the trajectory directions.
![scvelo_EHT_genes_velocity](https://user-images.githubusercontent.com/47226253/115311866-5f122100-a13e-11eb-86a7-b4ba85ed9d60.png)

The two things that I am considering before taking a decision about using this data or not is 1) If it would be possible to bias the HVG calculation to use EHT markers and if that would change my directionality 2) If it would be useful to do an imputation on unspliced counts based on similarity computed on spliced counts with a software like MAGIC and recalculate my velocities.

Thank you for putting together this amazing tool and sorry for the long message! I would really appreciate if you could give me your oppinion about this data  before I take a final conclusion as I am considering that maybe the nature of my dataset will make it not possible to use these results.

Please, let me know if there is any further information I could provide.
Thank you!

Best, 
Gabriel

**Comment by VolkerBergen** [comment]
Hi Gabriel, 

good timing to ask that Q as we're currently wrapping a perspectives paper highlighting some of the limitations in RNA velocity estimation, particularly in hematop. 

Generally, I'd advise to be extra careful with conclusions in hematop., as we've observed all kinds of interesting effects, including transcriptional boost, time-dependent decay rates etc. (e.g. in erythroids) that are not correctly captured by current velocity models. 

After all, you'd have to make sure you got some informative genes for each cell type that are supporting the inferred directionality. Hence, if you send me phase portraits of top likelihood genes (say top20) I can tell you more about applicability.

**Reply by GabrielBaldissera** [reply]
Hi,

Thank you for this super fast reply!! I'm glad to hear that and very excited to read that paper anytime soon!
Right! We're trying to be very careful drawing out conclusions out of this data.

Here is the plot of the top 20 likelihood genes:
![image](https://user-images.githubusercontent.com/47226253/115428917-dee8cb80-a1d0-11eb-9e90-508b9395b6ce.png)

**Reply by VolkerBergen** [reply]
It mostly picked those from cluster 5 if I get the color right, showing quite some indication that kinetics are actually supporting the direction from green (cluster 4) towards the tip (in cluster 5). For the rest, you could show the portraits stratified by celltype, following the last line in the [tutorial](https://scvelo.readthedocs.io/DynamicalModeling.html).

**Reply by GabrielBaldissera** [reply]
Right!
I have ploted 5 genes for each celltype following the tutorial:
![image](https://user-images.githubusercontent.com/47226253/115435131-8406a280-a1d7-11eb-8967-5cd89b8a9c13.png)

![image](https://user-images.githubusercontent.com/47226253/115435162-8bc64700-a1d7-11eb-8df3-3ee8571d810b.png)

![image](https://user-images.githubusercontent.com/47226253/115435177-91239180-a1d7-11eb-9556-1c39d4196adf.png)

![image](https://user-images.githubusercontent.com/47226253/115435201-984a9f80-a1d7-11eb-83f8-951d592f1ca3.png)
![image](https://user-images.githubusercontent.com/47226253/115435216-9da7ea00-a1d7-11eb-8976-8c3b7c6b6c19.png)
![image](https://user-images.githubusercontent.com/47226253/115435237-a1d40780-a1d7-11eb-8f5c-b86e2de12061.png)
![image](https://user-images.githubusercontent.com/47226253/115435249-a5678e80-a1d7-11eb-9c2f-1094cd451936.png)
![image](https://user-images.githubusercontent.com/47226253/115435262-a993ac00-a1d7-11eb-88d8-09f18aab3017.png)
![image](https://user-images.githubusercontent.com/47226253/115435340-ba442200-a1d7-11eb-9ce4-1587087caef5.png)
![image](https://user-images.githubusercontent.com/47226253/115435363-bf08d600-a1d7-11eb-82b2-1c8f3c8e1b88.png)

**Reply by VolkerBergen** [reply]
Several ambiguous genes (slope, no curvature), that are mostly being picked up as up-regulation, but could equally be interpreted as down-regulation. Overall, they indicate the process to be rooted in either cluster 0 or 1 (orange/red), differentiating towards 5 and 8. So the left part is supported by the kinetics, the right part looks too noise to make unambiguous conclusions.

**Reply by GabrielBaldissera** [reply]
Right.

My understanding was in the same direction that the left part makes sense according to EHT while the right part of the plot is very unclear. We were curious to see the behavior of cluster 6 and what makes it different to cluster 4 but apparently it contributes to the differentiation towards cluster 5. Our first aim was to understand if all those clusters in the right would go through EHT. Based on other analysis, we think that 2 and 1 could do it.
 
I have ploted the terminal states as well. These plots were also a source of discussion to us. If I'm reading this correctly, the root areas are being found in cluster 3 and 6 while the endpoints are in clusters 5, 8 and 1. If that was the case, we would understand that the root is being misread. The EHT process is a exemption is the biological development. The common developmental process would be Stem-> Somatic but in EHT we have a Somatic->Stem->Somatic. I would therefore expect the root value to be stronger in cluster in the right like cluster 1.

![image](https://user-images.githubusercontent.com/47226253/115442574-2b87d300-a1e0-11eb-893e-5fadbf9b64a9.png)
![image](https://user-images.githubusercontent.com/47226253/115442590-304c8700-a1e0-11eb-9d79-6e13839a3f03.png)

**Reply by GabrielBaldissera** [reply]
Thank you for your help @VolkerBergen!

I have decided not to use the data from this experiment. I will be very excited to read the new paper you are wrapping up once it's released.

Best,
Gabriel
